src/charm.py

```diff
@@ -213,28 +232,31 @@ def init_server(self):
             {
                 "state": "started",
                 "unified": "true" if self.tls.upgrading else "",
-                "quorum": self.cluster.quorum or "",
+                "quorum": self.cluster.quorum,
             }
         )
 
-    def config_changed(self):
+    def config_changed(self) -> bool:
         """Compares expected vs actual config that would require a restart to apply."""
-        properties = (
-            pull(container=self.container, path=self.zookeeper_config.properties_filepath).split(
-                "\n"
-            )
-            or []
-        )
+        try:
+            properties = pull(
+                container=self.container, path=self.zookeeper_config.properties_filepath
+            ).split("\n")
+        except PathError:
+            properties = []
```

Comment:
**question/thought/non-blocking:** This looks like a really sensible addition. Where in the event lifecycle were you seeing `PathError` crop up?

Depending on where it is, it might not be great to default to an empty list, because then `config_changed` will just write the whole thing. 

That said, the previous example would `raise`, also bad, so I wonder if there's a potential bug hiding here that might need more attention in another PR.

Reply:
I believe it was in the unittests. When doing this addition I was also looking into the `zookeeper-operator` repo. There we have a `safe_get_file` ([here](https://github.com/canonical/zookeeper-operator/blob/main/src/charm.py#L224)) that - if the files does not exists - returns None, and then the value is overridden with empty list. 

The `safe_get_file` seems a nice touch to be used here as well, but I felt it would have required a bit larger refactoring (to use `safe_get_file` consistently across the entire repo) that was outside of the scope of the PR. IMHO, that should go in the large refactoring to have common funcionalities in both k8s and vm, only the `safe_get_file` implementation is different between k8s and vm